ethereum/spec/src/property-test/java/tech/pegasys/teku/spec/datastructures/execution/versions/capella/WithdrawalPropertyTest.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright ConsenSys Software Inc., 2022
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.spec.datastructures.execution.versions.capella;
+
+import static tech.pegasys.teku.spec.datastructures.util.PropertyTestHelper.assertDeserializeMutatedThrowsExpected;
+import static tech.pegasys.teku.spec.datastructures.util.PropertyTestHelper.assertRoundTrip;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import net.jqwik.api.ForAll;
+import net.jqwik.api.Property;
+import tech.pegasys.teku.spec.SpecMilestone;
+import tech.pegasys.teku.spec.propertytest.suppliers.DataStructureUtilSupplier;
+import tech.pegasys.teku.spec.util.DataStructureUtil;
+
+public class WithdrawalPropertyTest {
+
+  @Property
+  void roundTrip(@ForAll(supplier = WithdrawalSupplier.class) final Withdrawal withdrawal)
+      throws JsonProcessingException {
+    assertRoundTrip(withdrawal);
+  }
+
+  @Property
+  void deserializeMutated(
+      @ForAll(supplier = WithdrawalSupplier.class) final Withdrawal withdrawal,
+      @ForAll final int seed) {
+    assertDeserializeMutatedThrowsExpected(withdrawal, seed);
+  }
+
+  public static class WithdrawalSupplier extends DataStructureUtilSupplier<Withdrawal> {
```

Comment:
I'd pull these out to a separate file so it's easier to reuse if we need it in the future.